testsuite/testsuite-messaging/src/test/java/org/wildfly/swarm/messaging/test/MessagingTest.java

```diff
@@ -0,0 +1,31 @@
+package org.wildfly.swarm.messaging.test;
+
+import static org.fest.assertions.Assertions.assertThat;
+
+import java.io.File;
+import java.nio.file.Files;
+
+import org.jboss.shrinkwrap.api.ShrinkWrap;
+import org.jboss.shrinkwrap.api.exporter.ZipExporter;
+import org.junit.Test;
+import org.wildfly.swarm.fractions.FractionUsageAnalyzer;
+import org.wildfly.swarm.spi.api.JARArchive;
+
+public class MessagingTest {
+
+    @Test
+    public void testFractionMatching() throws Exception {
+	    JARArchive archive = ShrinkWrap.create(JARArchive.class);
+	    archive.addClass(MyTopicMDB.class);
+	    FractionUsageAnalyzer analyzer = new FractionUsageAnalyzer();
+
+	    final File out = Files.createTempFile(archive.getName(), ".war").toFile();
```

Comment:
Actually, one comment: this file should be deleted at the end of the test. `try/finally` looks like the best choice.